conans/client/installer.py

```diff
@@ -325,17 +305,44 @@ def _raise_missing(self, missing):
             return
 
         missing_prefs = set(n.pref for n in missing)  # avoid duplicated
-        for pref in sorted(missing_prefs):
+        missing_prefs = list(sorted(missing_prefs))
+        for pref in missing_prefs:
             self._out.error("Missing binary: %s" % str(pref))
         self._out.writeln("")
 
-        # Raise just the first one
+        # Report details just the first one
         node = missing[0]
         package_id = node.package_id
         ref, conanfile = node.ref, node.conanfile
         dependencies = [str(dep.dst) for dep in node.dependencies]
-        raise_package_not_found_error(conanfile, ref, package_id, dependencies,
-                                      out=conanfile.output, recorder=self._recorder)
+
+        settings_text = ", ".join(conanfile.info.full_settings.dumps().splitlines())
+        options_text = ", ".join(conanfile.info.full_options.dumps().splitlines())
+        dependencies_text = ', '.join(dependencies)
+        requires_text = ", ".join(conanfile.info.requires.dumps().splitlines())
+
+        msg = textwrap.dedent('''\
+            Can't find a '%s' package for the specified settings, options and dependencies:
+            - Settings: %s
+            - Options: %s
+            - Dependencies: %s
+            - Requirements: %s
+            - Package ID: %s
+            ''' % (ref, settings_text, options_text, dependencies_text, requires_text, package_id))
+        conanfile.output.warn(msg)
+        self._recorder.package_install_error(PackageReference(ref, package_id),
+                                             INSTALL_ERROR_MISSING, msg)
+        missing_pkgs = ", ".join([str(pref.ref) for pref in missing_prefs])
```

Comment:
Better looking if each name is enclosed in quotes:

```suggestion
        missing_pkgs = "', '".join([str(pref.ref) for pref in missing_prefs])
```